### PR TITLE
ipa320 default gitignore

### PR DESCRIPTION
**Pull request by fmessmer**
@ipa-mig @ipa-fmw @ipa-nhg @ipa-bnm 
How do you like this one?
related to https://github.com/ipa320/cob_calibration_data/pull/118

I'm not sure about `bin` and `lib` as we sometimes have files in these folder which we don't want to be gitignored...e.g. cob_extern....
I think we could remove `build`, `bin` and `lib` as `catkin` does not create those folders in the source space anymore...

### CODE REVIEW

File: .gitignore
Comment (mgruhler) [inline .gitignore:42]: yay, I especially like this one!

File: .gitignore
Comment (mgruhler) [inline .gitignore:6]: I would remove `build`, as this is definitely not required. However, I'd keep `bin` and `lib`. This does not hurt and actually forces us to think about if what we are doing makes sense.

As soon as you have added something in those folders (with `git add -f`),  those files are tracked regularly.

**Review by mgruhler — CHANGES_REQUESTED** [review]
I would add:

```
# VI/VIM
*.swp
```

Also, one could think about libraries and compiled objects (`*.slo, *.lo, *.o, *.so, *.dylib, *.lai, *.la, *.a`) but I don't think we need those.

**Review by benmaidel — APPROVED** [review]

**Comment by fmessmer** [comment]
considered @ipa-mig's comments
@ipa-fmw please provide feedback...

After approval, I will provide PRs to the relevant repositories

**Review by mgruhler — APPROVED** [review]

**Comment by fmessmer** [comment]
Here is a list of candidates for the default .gitignore:

### bringup packages
 - cob_android
 - cob_calibration_data
 - cob_command_tools
 - cob_common
 - cob_control
 - cob_docking
 - cob_driver
 - cob_environments
 - cob_extern
 - cob_gazebo_plugins
 - cob_hand
 - cob_manipulation
 - cob_navigation
 - cob_perception_common
 - cob_robots
 - cob_simulation
 - cob_substitute
 - schunk_modular_robotics
 - schunk_robots
 - sick_flexisoft

### msh packages
 - cob_people_perception
 - ipa_navigation
 - ipa_navigation_common
 - ipa_navigation_driver
 - ipa_navigation_localization
 - ipa_navigation_perception
 - ipa_navigation_planning
 - ipa_navigation_sandbox
 - msh
 - msh_common
 - msh_gui
 - msh_voice_control


Did I forget something? Should any of the packages above **NOT** use the default .gitignore?

**Comment by mgruhler** [comment]
all navigation packages could use the new one. 

I guess the rest should be fine as well. I'm not sure about msh, though.

**Comment by mgruhler** [comment]
Still waiting for @ipa-fmw's feedback, though?

**Comment by fmessmer** [comment]
Yes, I'd like to see approval of @ipa-fmw and @ipa-nhg for this before creating 30 PRs...

**Review by floweisshardt — APPROVED** [review]
looks good to me. OK to roll out to all repos.

**Comment by fmessmer** [comment]
Well, I'll go on with this now...I have three approvals...

**Comment by fmessmer** [comment]
updated all repos from the list above!

**Pull request merged**
